docs/contributors/folder-structure.md

```diff
@@ -0,0 +1,192 @@
+    │
+    ├── .wordpress-org/*
+    │   Assets of the WooCommerce Blocks plugin hosted on the WordPress plugin
+    │   repository.
+    │
+    ├── assets/css/*
+    │   The SCSS files of the WooCommerce Blocks plugin.
+    │
+    ├── assets/js/*
+    │   The React components of the WooCommerce Blocks plugin.
+    │
+    ├── assets/js/atomic
+    │   The atomic components such as product title, product rating, product
+    │   image, etc. These atomic components are used by various blocks.
+    │
+    ├── assets/js/base
+    │   Functionality to ensure that the frontend does not get too bloated.
+    │
+    ├── assets/js/blocks-registry
+    │   Files that allows developers to connect their extensions to this plugin.
+    │
+    ├── assets/js/blocks
+    │   The main blocks of the WooCommerce Blocks plugin such as the Active 
+    │   Filters Block.
+    │
+    ├── assets/js/data
+    │   Functionality to store data using Redux and wp.data.
+    │
+    ├── assets/js/editor-components
+    │   Editor components such as the block-title component.
+    │
+    ├── assets/js/extensions
+    │   TypeScript files to allow Google Analytics tracking of specific events
+    │   such as active payment method and removing cart items.
+    │
+    ├── assets/js/filters
+    │   Filter such as excluding the checkout draft from Google Analytics.
+    │
+    ├── assets/js/hocs
+    │   The Higher Order Components of the WooCommerce Blocks plugin.
+    │
+    ├── assets/js/icons
+    │   The SVG icons of the WooCommerce Blocks plugin.
+    │
+    ├── assets/js/legacy
+    │   The legacy code of the WooCommerce Blocks plugin.
+    │
+    ├── assets/js/middleware
+    │   The middleware code to handle Store API calls.
+    │
+    ├── assets/js/payment-method-extensions
+    │   Functionality for the payment options such as PayPal and Stripe.
+    │
+    ├── assets/js/previews
+    │   The previews of various components such the All Products Block.
+    │
+    ├── assets/js/settings
+    │   Functionality to view the settings in the frontend. 
+    │
+    ├── assets/js/shared
+    │   Shared components of the WooCommerce Blocks plugin.
+    │
+    ├── assets/js/types
+    │   TypeScript definitions of the WooCommerce Blocks plugin.
+    │
+    ├── assets/js/utils
+    │   Shared utilities of the WooCommerce Blocks plugin.
+    │
+    ├── bin/*
+    │   Set of scripts used to build the WordPress packages.
+    │
+    ├── docs/*
+    │   Set of documentation pages of the WooCommerce Blocks plugin.
+    │
+    ├── images
+    │   Images for the payment options and the previews.
+    │
+    ├── packages
+    │   Source code of the WordPress packages. Packages can be:
+    │    - Production JavaScript scripts and styles loaded on WordPress
+    │      and the WooCommerce Blocks plugin or distributed as npm packages.
+    │    - Development tools available on npm.
+    │
+    ├── patches
+    │   The patches of the WooCommerce Blocks plugin.
```

Comment:
```suggestion
    │   Patches for 3rd party scripts applied when installing dependencies.
```

Reply:
Thanks for this suggestion, @mikejolley!